src/lib/components/forms/DocumentUpload.svelte

```diff
@@ -0,0 +1,458 @@
+      type: "error",
+      status: process_response.status,
+      error: await process_response.text(),
+    };
+  }
+</script>
+
+<script lang="ts">
+  import type {
+    Access,
+    Document,
+    DocumentUpload,
+    OCREngine,
+    Project,
+  } from "$lib/api/types";
+
+  import { applyAction } from "$app/forms";
+  import { filesize } from "filesize";
+  import { afterUpdate } from "svelte";
+  import { _ } from "svelte-i18n";
+  import { File16, File24, Upload16, XCircleFill24 } from "svelte-octicons";
+
+  import { page } from "$app/stores";
+
+  import Button from "../common/Button.svelte";
+  import Empty from "../common/Empty.svelte";
+  import Field from "../common/Field.svelte";
+  import FieldLabel from "../common/FieldLabel.svelte";
+  import Flex from "../common/Flex.svelte";
+  import Premium from "../common/Premium.svelte";
+
+  import AccessLevel from "../inputs/AccessLevel.svelte";
+  import Dropzone from "../inputs/Dropzone.svelte";
+  import FileInput from "../inputs/File.svelte";
+  import Language from "../inputs/Language.svelte";
+  import Select, { unwrap } from "../inputs/Select.svelte";
+  import Switch from "../inputs/Switch.svelte";
+  import Text from "../inputs/Text.svelte";
+
+  import * as documents from "$lib/api/documents";
+  import { DEFAULT_LANGUAGE, DOCUMENT_TYPES } from "@/config/config.js";
+  import { isSupported } from "@/lib/utils/validateFiles";
+
+  let files: File[] = [];
+  let projects: Project[] = [];
+
+  let loading = false;
+  let uploader: HTMLInputElement;
+
+  let fileDropActive: boolean;
+
+  const ocrEngineOptions = [
+    {
+      value: "tess4",
+      label: "Tesseract",
+      help: $_("uploadOptions.tesseract"),
+    },
+    {
+      value: "textract",
+      label: "Textract",
+      help: $_("uploadOptions.textract"),
+    },
+  ];
+
+  let ocrEngine = ocrEngineOptions[0];
+
+  $: csrf_token = $page.data.csrf_token;
+  $: projects = $page.data.projects.results;
+
+  $: total = files.reduce((t, file) => {
+    return t + file.size;
+  }, 0);
+
+  function addFiles(filesToAdd: FileList) {
+    files = files.concat(Array.from(filesToAdd).filter(isSupported));
+  }
+
+  function removeFile(index: number) {
+    files = files.filter((f, i) => i !== index);
+  }
+
+  function formatFileType(filetype: string) {
+    if (filetype && filetype.includes("/")) {
+      return filetype.split("/")[1];
+    }
+    return "unknown";
+  }
+
+  function filenameToTitle(filename: string): string {
+    const [name, ...ext] = filename.split(".");
+    return name.replace(/_/g, " ");
+  }
+
+  // handle uploads client side instead of going through the server
+  async function onSubmit(e: SubmitEvent) {
+    loading = true;
+    const form = e.target as HTMLFormElement;
+    const fd = new FormData(form);
+
+    const result = await upload(fd, csrf_token);
+
+    // send data up
+    await applyAction(result);
+
+    if (result.type === "success") {
+      form.reset();
+      files = [];
```

Comment:
After the upload succeeds, could we send users to the manager with a preset search for their user documents sorted by most recently uploaded?

Reply:
We can. Right now, we split the view of a user's documents into public and private. We probably want to put those back together if we're sending you back to your recent documents.

Reply:
:+1: We can decide where to send the user based on the access level of their upload, yeah? Like:

`if (fd.access === 'public') { goto(/* public user docs */ ) } else { goto(/* private user docs */) }`

Reply:
Added this in, with the caveat that the new documents often aren't in the Solr index yet, so they don't show up until refresh. I added an issue (#512) to deal with this, but I don't think it should block us from merging this.